### ISSUE TITLE
Biblatex export. Date parses incorrectly with year '1000' when source Zotero field is in datetime format.

### ISSUE THREAD

**Issue by JohnLukeBentley** [retorquere/zotero-better-bibtex#515]
In Zotero I have an item with (among other fields):

    Date: 2016-06-06T15:26:37+1000

It probably doesn't matter but I also have...

    Url: [some url]
    Accessed: 2017-12-06 18:40:14 [Fictional, being a future date, just to help with differentiation]

I export (actually Quick Copy) using "Better Biblatex", the result is

```
@article{barker_1000_swiss,
  author = {Barker, Anne},
  date = {1000-06-06},
  title = {Swiss voters say no to guaranteed free money},
  urldate = {2017-12-06},
  journaltitle = {ABC News},
  langid = {australian},
  timestamp = {2016-06-06T08:50:13Z},
  abstract = {Voters in Switzerland overwhelmingly reject a proposal to introduce a guaranteed monthly income for all citizens.},
  url = {http://www.abc.net.au/news/2016-06-06/swiss-voters-reject-basic-income-proposal/7481672},
  rights = {http://www.abc.net.au/conditions.htm\#UseOfContent}
}
```

The biblatex `date` field incorrectly parses the Zotero date field as

    1000-06-06

It should be

     2016-06-06

A reminder that biblatex does not currently support times in the `date` field. There are [ongoing discussions](https://github.com/plk/biblatex/issues/427#issuecomment-223897298) about whether biblatex should support times (at least not complain if times end up in the date field and other date type fields).

But perhaps, for the moment in Zotero-Better-Bibtex, parsing `2016-06-06T08:50:13Z` as `2016-06-06` (rather than `1000-06-06` or `2016-06-06T08:50:13Z`) would be the way to go.

**Comment by retorquere** [comment]
Can you right-click on that reference and select "send BBT error"? That will give me that reference as a test case (click through the dialogs and paste the red ID here).

**Comment by JohnLukeBentley** [comment]
Report ID: EFSF3WC6

The other day I inadvertently sent you a bunch of Reports, which you can ignore.

**Comment by retorquere** [comment]
No worries, I never look at reports unless an ID is posted here, and reports are deleted after a week automatically.

**Comment by retorquere** [comment]
Can you try https://github.com/retorquere/zotero-better-bibtex/releases/download/builds/zotero-better-bibtex-1.6.61-circle-2532.xpi ?

**Comment by JohnLukeBentley** [comment]
It works.

I've only tried my one test case. I'm going to assume that takes care of it and reopen the issue only if I come across another exotic, bug triggering, case in production.

Thanks Emiliano, I hope that afforded you a nice break ;)

**Comment by retorquere** [comment]
It also passes all my other tests, or I couldn't have gotten you that test
version. I can't publish test or release versions unless all tests are
green.

Yeah, I could use the break. I have my defense this week, and it's been
occupying my every waking moment (of which there are currently more than is
generally considered healthy). These small fixes are great procrastination
targets. Not too big, but a break nonetheless.

**Comment by JohnLukeBentley** [comment]
Yeah, regression testing. A very good thing to do.

Phd defense? Good luck with that. 

If I send you further issues I'll know there may be good reasons for ignoring them for a while.

Naturally I'm instantly curious about what your thesis is - but you should get back to it, at once! :)

**Comment by retorquere** [comment]
No, just a masters degree in phil of science, tech and society. My thesis (such as it is) is about theories of well-being, and hedonism as particular, as an analytic device for the risks and benefits of technologically mediated friendships (which is just a fancy way of saying "is reddit good for you? news at 11")

**Comment by JohnLukeBentley** [comment]
Well, shit. That's interesting.

Although no longer in the academy my specific interest was, and is, in what I prefer to call "prudential theories", more or less "theories of well-being". 

I'm interested in various meta issues: e.g. properly distinguishing the prudential good from the moral good and the good, all things considered. But also very much in defending particular prudential claims: e.g. "Monosexuality (normally called 'monogamy') is prudentially bad". Indeed part of my motive for wanting to properly sort out the meta issues is to furnish us all with sharper tools to argue over particular prudential claims.

After you receive your thesis result perhaps you could post it, or a TL:DR; version of it, to reddit.

Entirely relevant to we redditors. I was only today reflecting on Snowden's cheeky comment ...
> "people say I live in Russia, but that's actually a little bit of a misunderstanding. I live on the Internet." https://en.wikipedia.org/wiki/Edward_Snowden

... and the extent to which it applies to myself. 

So your thesis is relevant to my interests :)

**Comment by retorquere** [comment]
> Although no longer in the academy my specific interest was, and is, in
what I prefer to call "prudential theories", more or less "theories of
well-being".
>
> I'm interested in various meta issues: e.g. properly distinguishing the
prudential good from the moral good and the good, all things considered.

That was an important distinction for my work, since I was arguing in favor
of prudential hedonism. Most other theories of well-being seem very happy
to conflate the two, but hedonism sort of can't afford to.

Not a hedonist myself, although 4 years down the road I'm harder pressed
than ever to say where I stand. I used to be very partial to virtue theory
of well-being, until I learned about its problems. Now I sort of just wish
I could be a virtue theorist of well-being. Certainly not an objective
lister, nor a desire satisfactionist. Unfortunately, one of my opponents
*is*, so the defense is going to be interesting. Assuming you're not me
that is. In that case it leans more towards terrifying.

> But also very much in defending particular prudential claims: e.g.
"Monosexuality (normally called 'monogamy') is prudentially bad".

Let's see, I smell a touch of hedonism there, perhaps a little desire
satisfactionism, a little x-phi or psychology, if you're on the empirical
side, otherwise, if you're on the normative-conceptual side, I'd say some
derivative of objective list.

> Indeed part of my motive for wanting to properly sort out the meta issues
is to furnish us all with sharper tools to argue over particular prudential
claims.

My actual interest is old-school Popper-style phil of science, phil of
expertise, epistemology, and phil of mind, but the faculty I was admitted
to doesn't do any of those (at least, it does none of those seriously). I
took this project because it was valuable research to someone else, as the
previous project (phil of sci/expertise, ranging from Popper to Foucault to
STS (ptooey) to James to... anyhow, a million fields) that was valuable to
me spun out of control. Plus, the test ought to be whether I can reason
philosophically, not whether I can make a passionate plea for my pet
worldview. Not a fan of hedonism, but a plausible case can be made.

Not that this one hasn't, but at least it allowed me to get to a place
where I could say "screw all this noise, pick a date, this is what I'm
going to defend".

> After you receive your thesis result perhaps you could post it, or a
TL:DR; version of it, to reddit.

Er... two people on this earth have read the thing so far. Two more will. I
reckon that's about 4 people too many.

>> "people say I live in Russia, but that's actually a little bit of a
misunderstanding. I live on the Internet."
https://en.wikipedia.org/wiki/Edward_Snowden

Also a point I argue - that "online" is a place no less real than offline
places, and that actions cannot be deemed less real merely because they
happen there.
